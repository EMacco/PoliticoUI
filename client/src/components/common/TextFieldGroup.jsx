import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const TextFieldGroup = ({
  id,
  placeholder,
  value,
  className,
  type,
  name,
  onChange,
  icon,
  error
}) => {
  return (
    <Fragment>
      <div className="input-with-icon" style={{ position: 'relative' }}>
        <img src={icon} />
        <input
          type={type}
          className={className}
          placeholder={placeholder}
          value={value}
          name={name}
          onChange={onChange}
          id={id}
          required
        />
      </div>
      {error && <div className="errorDiv">{error}</div>}
    </Fragment>
  );
};

TextFieldGroup.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  icon: PropTypes.string,
  error: PropTypes.string
};

TextFieldGroup.defaultProps = {
  type: 'text',
  className: 'signin-text-field'
};

export default TextFieldGroup;
