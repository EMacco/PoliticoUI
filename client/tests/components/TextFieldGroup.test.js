import React from 'react';
import TextFieldGroup from '@components/common/TextFieldGroup';

const props = {
  id: '1',
  name: 'email',
  placeholder: 'Email',
  value: '',
  className: '',
  type: 'input',
  onChange: () => {},
  icon: '',
  error: 'Email is required'
};

const setup = () => {
  const wrapper = shallow(<TextFieldGroup {...props} />);
  return wrapper;
};

describe('<TextFieldGroup />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it('should render the TextFieldGroup component correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('TextFieldGroup')).toBeTruthy();
  });
});
