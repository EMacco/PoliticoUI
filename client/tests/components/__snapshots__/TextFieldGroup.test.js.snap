// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFieldGroup /> should render the TextFieldGroup component correctly 1`] = `
<Fragment>
  <div
    className="input-with-icon"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <img
      src=""
    />
    <input
      className=""
      id="1"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="input"
      value=""
    />
  </div>
  <div
    className="errorDiv"
  >
    Email is required
  </div>
</Fragment>
`;
